javarosa/src/main/java/org/javarosa/xform/util/CalendarUtils.java

```diff
@@ -359,14 +359,19 @@ public static UniversalDate incrementYear(UniversalDate date) {
                 year,
                 month,
                 day,
-                toMillisFromJavaEpoch(year, month, day, date.millisFromJavaEpoch % UniversalDate.MILLIS_IN_DAY)
+                toMillisFromJavaEpoch(year, month, day)
         );
     }
 
-    public static long toMillisFromJavaEpoch(int year, int month, int day, long millisOffset) {
+    public static long toMillisFromJavaEpoch(int year, int month, int day) {
+        return toMillisFromJavaEpoch(year, month, day, TimeZone.getDefault());
+    }
+
+    public static long toMillisFromJavaEpoch(int year, int month, int day, TimeZone currentTimeZone) {
         int daysFromMinDay = countDaysFromMinDay(year, month, day);
         long millisFromMinDay = daysFromMinDay * UniversalDate.MILLIS_IN_DAY;
-        return millisFromMinDay + MIN_MILLIS_FROM_JAVA_EPOCH + millisOffset;
+        int timezoneOffsetFromGMT = currentTimeZone.getOffset(millisFromMinDay);
+        return millisFromMinDay + MIN_MILLIS_FROM_JAVA_EPOCH - timezoneOffsetFromGMT;
```

Comment:
The idea is to normalize the 'milliseconds since epoch' to not include timezone info, since we are representing a date not a datetime. I guess I did half of this in my last fix; the half when converting from millis to a date. The issue is that I didn't do the other half: removing the timezone offset when converting a date to 'milliseconds since epoch'. That is what this PR does.